spec/worker_spec.rb

```diff
@@ -65,38 +65,22 @@
       end
     end
 
-    [PG::UnableToSend, PG::ConnectionBad].each do |exception|
-      context "with a #{exception} exception" do
-        before do
-          original_execute = QC.default_conn_adapter.method(:execute)
-          allow(QC.default_conn_adapter).to receive(:execute) do |*args|
-            if args.first == 'ROLLBACK'
-              raise exception
-            else
-              original_execute.call(*args)
-            end
-          end
-        end
+    context 'when PG connection reaped during a job' do
+      before { Jobs::Tests::ConnectionReapedTestJob.enqueue_perform }
 
-        it 'retries without incrementing retries' do
-          job_type.enqueue_perform(true)
-          Timecop.freeze do
-            worker.work
-            expect(failed_queue.count).to eq 0
-            QueueClassicMatchers::QueueClassicRspec.find_by_args('low', 'Jobs::Tests::LockedTestJob._perform', [true]).first['remaining_retries'].should eq "5"
-          end
+      it 'retries without incrementing retries' do
+        Timecop.freeze do
+          worker.work
+          expect(failed_queue.count).to eq 0
+          QueueClassicMatchers::QueueClassicRspec.find_by_args('low', 'Jobs::Tests::ConnectionReapedTestJob._perform', []).first['remaining_retries'].should eq "5"
         end
+      end
 
-        context 'with retries disabled for the class' do
-          let(:job_type) { Jobs::Tests::TestJobNoRetry }
-
-          it 'enqueues the failed queue' do
-            job_type.enqueue_perform(true)
-            Timecop.freeze do
-              worker.work
-              expect(failed_queue.count).to eq 1
-            end
-          end
+      it 'ensures to rollback' do
+        allow(QC.default_conn_adapter).to receive(:execute).and_call_original
```

Comment:
no, without this, we cannot execute any `eunuque` or get job stuff because of the next constraint.